### PR TITLE
Improve Testing Coverage for Menu

### PR DESCRIPTION
**Pull request by esilverm**
Signed-off-by: Evan Silverman <es4753@nyu.edu>

#### What does this PR do?

`onPreviousMenuItem` had little coverage so I introduced tests to open the Menu with the `up` key and to navigate backwards through the Menu with `shift + tab` keys. Also added a test for bottom aligned drop.

#### Where should the reviewer start?

`/src/js/components/Anchor/__tests__/Menu-test.js`

#### What testing has been done on this PR?

Run `yarn test` to ensure that all tests pass and view coverage

#### What are the relevant issues?

#4539

#### Screenshots (if appropriate)
Before (80.52%)
Screen Shot 2020-10-06 at 3 26 56 PM

After (97.40%)
Screen Shot 2020-10-06 at 4 00 07 PM

#### Do the grommet docs need to be updated?

No

#### Is this change backwards compatible or is it a breaking change?

Backwards Compatible

### CODE REVIEW

File: src/js/components/Menu/__tests__/Menu-test.js
Comment (jcfilben) [inline src/js/components/Menu/__tests__/Menu-test.js:374]: Can we put an assertion after the 'Up' keyDown to check that the Menu is being opened

File: src/js/components/Menu/__tests__/Menu-test.js
Comment (jcfilben) [inline src/js/components/Menu/__tests__/Menu-test.js:304]: Add assertions between keyDown's to make sure expected behavior is actually happening

**Review by jcfilben — COMMENTED** [review]
Nice coverage increase! Dropped a few comments below

File: src/js/components/Menu/__tests__/Menu-test.js
Comment (esilverm) [reply src/js/components/Menu/__tests__/Menu-test.js:374]: Good idea, about to push the changes to my fork

**Review by jcfilben — APPROVED** [review]
Thanks for adding the additional assertions. I looked over the snapshots as well and they look good to me

**Comment by esilverm** [comment]
I think I may be able to get a bit more coverage by implementing the tab + shift navigation on a menu where the control is at the bottom. I can work on adding this a bit later.

**Comment by esilverm** [comment]
I tried adding the test cases and it only improved branches by 3 so I decided to take a look into why it wasn't affecting the code. It looks like the [line I was trying to cover](https://github.com/grommet/grommet/blob/64758e5bb5bb3b3bc4d01b3fa65b31516bebe4a9/src/js/components/Menu/Menu.js#L177) is dependent on the arrow keys being used for changing focus of the elements. This was implemented in #3334

However, it looks like the [keyboard handlers](https://github.com/grommet/grommet/blob/64758e5bb5bb3b3bc4d01b3fa65b31516bebe4a9/src/js/components/Menu/Menu.js#L253) for changing focus of menu items with arrow keys were replaced in favor of using tab later on. This was implemented in #3464

I don't think the code to handle arrow key events was ever tackled after that rollback. If you want, I can set up an issue and attempt to tackle it. 

In my opinion this pr should be good to merge unless you want me to commit my changes which improve branch coverage by ~3%.

**Review by jcfilben — APPROVED** [review]
> I tried adding the test cases and it only improved branches by 3 so I decided to take a look into why it wasn't affecting the code. It looks like the line I was trying to cover is dependent on the arrow keys being used for changing focus of the elements.

That's awesome that you were able to uncover this through testing. I also appreciated the detailed explanation and links to the associated pull requests.

I think you can open another pull request for the keyboard handling and include the associated test you wrote for it there. This PR looks good to me and I think it is ready to be merged once @ShimiSun reviews and approves it.

**Review by ShimiSun — APPROVED** [review]

**Comment by ShimiSun** [comment]
Thank you for your contribution!

**Pull request merged**
